lib/util/logger/logger.go

```diff
@@ -12,18 +14,33 @@ import (
 
 type testingLog struct {
 	*testing.T
+	buf bytes.Buffer
 }
 
 func (t *testingLog) Write(b []byte) (int, error) {
 	t.Logf("%s", b)
-	return len(b), nil
+	return t.buf.Write(b)
+}
+
+func (t *testingLog) String() string {
+	return t.buf.String()
 }
 
 // CreateLoggerForTest creates a logger for unit tests.
 func CreateLoggerForTest(t *testing.T) *zap.Logger {
 	return zap.New(zapcore.NewCore(
 		zapcore.NewConsoleEncoder(zap.NewDevelopmentEncoderConfig()),
-		zapcore.AddSync(&testingLog{t}),
+		zapcore.AddSync(&testingLog{T: t}),
 		zap.InfoLevel,
 	)).Named(t.Name())
 }
+
+// CreateLoggerAndStringerForTest returns both the logger and its content.
+func CreateLoggerAndStringerForTest(t *testing.T) (*zap.Logger, fmt.Stringer) {
```

Comment:
I think it is identical as `CreateLoggerForTest`? Maybe:

1. define `CreateLoggerForTest` as `lg, _ := CreateLoggerAndStringerForTest()`.
2. replace `CreateLoggerForTest` with `CreateLoggerAndStringerForTest`